XUtils.c

```diff
@@ -236,6 +236,8 @@ size_t strnlen(const char* str, size_t maxLen) {
 #endif
 
 int xAsprintf(char** strp, const char* fmt, ...) {
+   *strp = NULL;
```

Comment:
I think there is a redundancy here. Since in one previous commit you are trying to ensure the caller has set `*strp` to NULL, and then here you set `*strp` to NULL once again. While this might seem harmless, I wonder if this can hide a potential memory leak. Like this hypothetical scenario:

```c
char* buf = xMalloc(len);
// ... without freeing buf ...
xAsprintf(&buf, "foo %d %d", x, y);
```

I would prefer that `xAsprintf` asserts that caller sets `*strp` to NULL, and not set it by itself.

Reply:
I was thinking of putting an assert in there, something like:

```
ASSERT(!*strp);
```

But dong that would change semantics for the caller AND change the behaviour on FreeBSD, which doesn't set `strp` on error (yes, we fail in that case, but nonetheless it's a different behaviour).

And regarding memory leaks: `valgrind` is your friend.